File: lib/models/contractor_member_link.dart
import 'package:flutter/foundation.dart';

@immutable
class ContractorMemberLink {
  final String contractorId;
  final String employeeId;
  final String? roleLabel;
  final bool isPrimary;

  const ContractorMemberLink({
    required this.contractorId,
    required this.employeeId,
    this.roleLabel,
    this.isPrimary = false,
  });
}

File: KaiFlow.Timesheets.Maui/Models/ContractorMemberLink.cs
using Supabase.Postgrest.Attributes;
using Supabase.Postgrest.Models;

namespace KaiFlow.Timesheets.Models;

[Table("contractor_member_links")]
public class ContractorMemberLink : BaseModel
{
    [PrimaryKey("id")]
    public Guid Id { get; set; }

    [Column("contractor_id")]
    public Guid ContractorId { get; set; }

    [Column("employee_id")]
    public Guid EmployeeId { get; set; }

    [Column("role")]
    public string? Role { get; set; }

    [Column("is_primary")]
    public bool IsPrimary { get; set; }

    [Column("company_id")]
    public Guid CompanyId { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }
}
